engine/core/src/main/java/org/datacleaner/configuration/RemoteServerConfigurationImpl.java

```diff
@@ -19,20 +19,54 @@
  */
 package org.datacleaner.configuration;
 
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.net.Socket;
+import java.net.URL;
 import java.util.ArrayList;
 import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
 import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+import org.datacleaner.job.concurrent.ScheduledTaskRunner;
+import org.datacleaner.job.concurrent.TaskListener;
+import org.datacleaner.job.concurrent.TaskRunner;
+import org.datacleaner.job.tasks.Task;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
  * Implementation of {@link RemoteServerConfiguration}.
  */
 public class RemoteServerConfigurationImpl implements RemoteServerConfiguration {
 
+    private static final Logger logger = LoggerFactory.getLogger(RemoteServerConfigurationImpl.class);
+    private static final int TEST_CONNECTION_TIMEOUT = 15 * 1000; // [ms]
+    private static final long ERROR_DELAY_MIN = 1;
+    private static final long OK_DELAY_MIN = 5;
+    private final Map<String, RemoteServerState> actualStateMap = Collections.synchronizedMap(new HashMap<>());
+    private ServerStatusTask serverStatusTask;
+    private ScheduledTaskRunner scheduledTaskRunner;
+    private List<RemoteServerStateListener> listeners = Collections.synchronizedList(new ArrayList());
```

Comment:
My IDE complains about a `Type safety: The expression of type List needs unchecked conversion to conform to List<RemoteServerStateListener>` here. Maybe add <> to the new ArrayList() object instanciation.